inc/templates.php

```diff
@@ -363,14 +448,21 @@ function hydrate_components( array $components ) {
 		$component->provide_context( get_template_context() );
 
 		// Recursively hydrate children.
-		if ( ! empty( $component->get_children() ) ) {
+		if ( ! is_string( $component ) && ! empty( $component->get_children() ) ) {
 			$component->set_children( hydrate_components( $component->get_children() ) );
 		}
 
 		// Reset context to where it was before hydration.
 		get_template_context()->reset();
 
-		$hydrated[] = $component->jsonSerialize();
+		// Convert text nodes to actual text notes.
+		// @todo there's _definitely_ a better way to do this, but certain
+		// Irving core functionality won't work without this hack.
+		if ( 'irving/text' === $component->get_name() ) {
```

Comment:
We might be able to handle this the same way we were talking about handling silent components, but cool to leave special handling for now.